api/nakadi-event-bus-api.yaml

```diff
@@ -1503,11 +1539,39 @@ definitions:
         type: string
         default: 'random'
 
+      compatibility_mode:
+        description: |
+          Compatibility mode provides a mean for event owners to evolve their schema, given changes respect the
+          semantics defined by this field.
+
+          It's designed to be flexible enough so that producers can evolve their schemas while not
+          inadvertently breaking existent consumers.
+
+          Once defined, the compatibility mode is fixed, since otherwise it would break a predefined contract,
+          declared by the producer.
+
+          List of compatibility modes:
+
+          - 'deprecated': Schema changes are not allowed. This is used only by legacy event types. This option is not
```

Comment:
> Would it <be> possible to default existing schemas to none and allow them to stay on none

It's not. The `none` mode allows for schema changes indiscriminately. That differs from the `deprecated` which doesn't allow changes. It means that consumers relying on fixed schemas would be broken by changes.

> or move up to compatible (if the schema in place meets the compatibility rules)

That's also not possible since currently, when `additionalProperties` is not used, the validator behaves as if it's `additionalProperties=true`. The new `compatible` mode changes that and makes sure that the behaviour will be like `additionalProperties=false`. So, even if the schema is "upgradable", the event validation behaviour would change, possibly breaking producers.

It will be possible to upgrade the event, but producers should make a PUT to the event type once they are sure it won't break their publishing routine.

> it could be renamed to "fixed" instead of "deprecated"

If it's just a matter of renaming, then I'm fine with that.

But the `deprecated` (or `fixed`) option won't be available for new event types. It's just a way to mark event types that are in the process of been moved to the new event types. All new event types should be either  `compatible` or `none`.